server/src/main/java/org/candlepin/dto/api/v1/InfoAdapter.java

```diff
@@ -0,0 +1,230 @@
+                    for (PermissionBlueprintDTO dto : blueprintDTOList) {
+                        blueprintInfoList.add(permissionBlueprintInfoAdapter(dto));
+                    }
+                }
+                return blueprintInfoList;
+            }
+        };
+    }
+
+    /**
+     * This method adapts the PermissionBlueprintDTO
+     * into PermissionBlueprintInfo object.
+     *
+     * @param source PermissionBlueprintDTO object
+     *
+     * @return PermissionBlueprintInfo object
+     */
+    public static PermissionBlueprintInfo permissionBlueprintInfoAdapter(PermissionBlueprintDTO source) {
+
+        return new PermissionBlueprintInfo() {
+
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public OwnerInfo getOwner() {
+                return ownerInfoAdapter(source.getOwner());
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public String getTypeName() {
+                return source.getType();
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public String getAccessLevel() {
+                return source.getAccess();
+            }
+        };
+    }
+
+    /**
+     * This method adapts the UserDTO
+     * into UserInfo object.
+     *
+     * @param source UserDTO object
+     *
+     * @return UserInfo object
+     */
+    public static UserInfo userInfoAdapter(UserDTO source) {
+        return new UserInfo() {
+
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public Date getCreated() {
+                return source.getCreated() != null ?
+                    new Date(source.getCreated().toInstant().toEpochMilli()) : null;
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public Date getUpdated() {
+                return source.getUpdated() != null ?
+                    new Date(source.getUpdated().toInstant().toEpochMilli()) : null;
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public String getUsername() {
+                return source.getUsername();
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public String getHashedPassword() {
+                return source.getPassword() == null ? source.getPassword() :
+                    Util.hash(source.getPassword());
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public Boolean isSuperAdmin() {
+                return source.getSuperAdmin();
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public Collection<? extends RoleInfo> getRoles() {
+                return null;
```

Comment:
why null?

Reply:
`UserDTO` does not have property `roles`